bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingManagerImpl.java

```diff
@@ -1258,15 +1251,35 @@ public void setEnabled(ThingUID thingUID, boolean enabled) {
         }
     }
 
+    private void persistThingEnableStatus(ThingUID thingUID, boolean enabled) {
+        if (storage == null) {
+            logger.debug("Cannot persist enable status of thing with UID {}. Persistent storage unavailable.", thingUID);
+            return;
+        }
+
+        logger.debug("Thing with UID {} will be persisted as {}.", thingUID, enabled ? "enabled." : "disabled.");
+        if (enabled) {
+            // Clear the disabled thing storage. Otherwise the handler will NOT be initialized later.
+            storage.remove(thingUID.getAsString());
+        } else {
+            // Mark the thing as disabled in the storage.
+            storage.put(thingUID.getAsString(), enabled);
```

Comment:
Since `enabled` is always false in this branch, I would prefer if we would use the literal value `false` in `storage.put()` here. That makes it more explicit that we only store the disabled status and not the enabled status.

Reply:
@YordanDZhelev pls also address this one, seems legitimate to me.